src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java

```diff
@@ -215,8 +215,25 @@ public void stop(@NonNull Throwable cause) throws Exception {
     @Override public void onResume() {
         try {
             if (state == null) {
-                Run<?, ?> run = getContext().get(Run.class);
-                LOGGER.fine(() -> "No ExecutorStepDynamicContext found for node block in " + run + "; perhaps loading from a historical build record, hoping for the best");
+                var flowNode = getContext().get(FlowNode.class);
+                LOGGER.fine(() -> "No ExecutorStepDynamicContext found for node block " + getContext() + "; will attempt to recover");
```

Comment:
So the previous message misleadingly implied that this could only happen with a build record created by an old plugin version, which is not the case—this is a normal situation. IOW
```suggestion
                LOGGER.fine(() -> "node block " + getContext() + " not yet scheduled, checking for an existing queue item");
```
would be clearer.